### PR TITLE
Added SpellRepository

### PR DESCRIPTION
**Pull request by david-buderus**

### CODE REVIEW

File: server/src/main/java/de/pnp/manager/component/Spell.java [hunk 1]
```diff
@@ -1,24 +1,60 @@
...
   private final String name;
+
+  /**
+   * The effect of this spell.
+   */
   private final String effect;
-  private final String type;
+
+  /**
+   * The cost to activate this spell.
```
Comment (bastihav) [inline server/src/main/java/de/pnp/manager/component/Spell.java]:
```suggestion
   * The cost to cast this spell.
```

File: server/src/main/java/de/pnp/manager/component/Spell.java [hunk 2]
```diff
@@ -1,24 +1,60 @@
...
+   */
   private final String cost;
+
+  /**
+   * The time needed to activate this spell.
```
Comment (bastihav) [inline server/src/main/java/de/pnp/manager/component/Spell.java]:
```suggestion
   * The time needed to cast this spell.
```

File: server/src/main/java/de/pnp/manager/component/Spell.java [hunk 3]
```diff
@@ -1,24 +1,60 @@
...
+   */
   private final String castTime;
+
+  /**
+   * The talents needed to activate this spell.
```
Comment (bastihav) [inline server/src/main/java/de/pnp/manager/component/Spell.java]:
```suggestion
   * The talents needed to cast this spell.
```

File: server/src/test/java/de/pnp/manager/server/database/SpellRepositoryTest.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            EPrimaryAttribute.INTELLIGENCE, EPrimaryAttribute.CHARISMA));
+
+    return new Spell(null, "Fireball", "Throw a fireball", "10 Mana", "0",
+        List.of(fireMagic), 2);
+  }
+
+  @Override
+  protected Spell createSlightlyChangeObject() {
+    Talent fireMagic = talentRepository.getByName(universe, "Fire Magic").stream().findFirst()
+        .orElseThrow();
```
Comment (bastihav) [inline server/src/test/java/de/pnp/manager/server/database/SpellRepositoryTest.java]: I do not really like that this method relies on the state of the other method. Please extract the `fireMagic` to a static attribute or similar.

**Review by bastihav — CHANGES_REQUESTED** [review]

File: server/src/test/java/de/pnp/manager/server/database/SpellRepositoryTest.java (hunk 4 above)
Comment (david-buderus) [reply server/src/test/java/de/pnp/manager/server/database/SpellRepositoryTest.java]: I moved it into an attribute. But IMHO it looks worse.
In the docs of `createSlightlyChangeObject` it is explict stated that you can access the state of `createObject`.

**Pull request merged**